### PR TITLE
[MRG] Fixes to CSP and example.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/4407

There was some unused parameters and other weirdness with the csp.

### CODE REVIEW

**Comment by agramfort** [comment]
LGTM +1 for MRG when CIs are happy

thx heaps @jaeilepp for catching this

**Comment by jaeilepp** [comment]
@jasmainak did most of the catching.

**Comment by Eric89GXL** [comment]
The `norm_trace defaults` warning shows up in the example, can it be avoided (preferably by using the new default `False`)?

**Comment by agramfort** [comment]
+1 for MRG when CIs are happy

**Comment by wmvanvliet** [comment]
Also, recent versions of Scikit-learn no longer have a `sklearn.lda.LDA` class. Do we want to update this while we're at it?

**Comment by kingjr** [comment]
> Also, recent versions of Scikit-learn no longer have a sklearn.lda.LDA
class. Do we want to update this while we're at it?

Yes it is in sklearn.discriminant_analysis, I think we can update it, or
even change it for a RidgeClassifier which is equivalent if you set the
regularization to 0 - that would prevent the sklearn incompatibility

File: examples/decoding/plot_decoding_csp_eeg.py [hunk 1]
```diff
@@ -73,22 +74,22 @@
...
 
-from sklearn.lda import LDA  # noqa
-from sklearn.cross_validation import ShuffleSplit  # noqa
-
-# Assemble a classifier
-lda = LDA()
-csp = CSP(n_components=4, reg=None, log=True)
+from sklearn.pipeline import Pipeline  # noqa
+from sklearn.discriminant_analysis import LinearDiscriminantAnalysis  # noqa
+from sklearn.model_selection import ShuffleSplit, cross_val_score  # noqa
```
Comment (Eric89GXL) [inline examples/decoding/plot_decoding_csp_eeg.py]: Might as well put these at the top, and then we can avoid the `noqa`

**Comment by Eric89GXL** [comment]
Thanks @jaeilepp

**Pull request merged**